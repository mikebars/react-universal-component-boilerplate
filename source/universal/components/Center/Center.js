/* @flow */

import styled from 'styled-components'

export const Center = styled.section`
  position: relative;

  text-align: center;

  top: 50%;

  transform: perspective(1px) translateY(-50%);
`

export default Center
