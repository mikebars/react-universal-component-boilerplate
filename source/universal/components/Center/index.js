/* @flow */

export { default } from './Center'
