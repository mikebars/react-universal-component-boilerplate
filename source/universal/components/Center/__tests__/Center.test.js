/* @flow */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount, shallow, render } from 'enzyme'

import { Center } from 'universal/components/Center/Center'

test('Center - snapshot test', () => {
  const props = {}

  const element = <Center {...props} />

  const tree = renderer.create(element).toJSON()

  expect(tree).toMatchSnapshot()
})

test('Center - enzyme shallow render', () => {
  const props = {}

  const element = <Center {...props} />

  const wrapper = shallow(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})

test('Center - enzyme full render', () => {
  const props = {}

  const element = <Center {...props} />

  const wrapper = mount(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})

test('Center - enzyme static render', () => {
  const props = {}

  const element = <Center {...props} />

  const wrapper = render(element)

  expect(wrapper).toBeDefined()
  expect(wrapper).toEqual(expect.anything())
})
