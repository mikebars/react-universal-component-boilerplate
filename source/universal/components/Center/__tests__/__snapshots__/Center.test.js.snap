// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center - snapshot test 1`] = `
<section
  className="Center-s1yx1lyo-0 lgKgCZ"
/>
`;
